faunadb-java/src/main/java/com/faunadb/client/types/time/HighPrecisionTime.java

```diff
@@ -59,52 +64,58 @@ private static int toInt(String value) {
    * @param microsToAdd microseconds to add to the initial timestamp
    * @param nanosToAdd  nanoseconds to add to the initial timestamp
    */
-  public HighPrecisionTime(Instant initialTime, int microsToAdd, int nanosToAdd) {
+  public static HighPrecisionTime fromInstant(Instant initialTime, long microsToAdd, long nanosToAdd) {
```

Comment:
Well. I stumbled upon a test where I asked for `Epoch(Value(30), MICROSECOND)` and I ended up having to calculate how many nanoseconds I had to pass into the `HighPrecisionTime`.

What about `fromInstant(Instant inst)`, `withNanos(Instant inst, long nanos)`, and `withMicros(Instant inst, long micros)`?